src/System/Components/PageLoadingBar.tsx

```diff
@@ -25,26 +27,43 @@ export const PageLoadingBar: React.FC<PageLoadingBarProps> = ({
     }
   }, [loadingState])
 
-  if (isServer) {
+  if (isServer || !isMounted) {
     return null
   }
 
   return (
-    <LoadingBar
-      ref={barRef as any}
-      loading={loading}
-      isComplete={isComplete}
-      backgroundColor="brand"
-      height={2}
-      top="1px"
-      left={0}
-      zIndex={100}
-      overflow="hidden"
+    <Box
       position="fixed"
-    />
+      top={0}
+      left={0}
+      zIndex={Z.pageLoadingBar}
+      width="100%"
+      height={3}
+    >
+      <LoadingBar
+        ref={barRef as any}
+        loading={loading}
+        isComplete={isComplete}
+        backgroundColor="brand"
+        height={2}
+        top="1px"
+        left={0}
+        width={isComplete ? barRef.current?.clientWidth : 0}
+        zIndex={100}
+        overflow="hidden"
+        position="fixed"
+        onAnimationEnd={() => {
+          if (isComplete) {
```

Comment:
What happens here if it's not complete?

Reply:
opacity starts at 0 to 1

Reply:
hard to get into this state, but will continue looking at it